### PR TITLE
feat: show history about completed runs in each cron workflow

### LINKED ISSUE #11706: UI - show history about completed runs in each cron workflow
# Summary

We can view completed cron-workflows in Workflow view. 
However, I suggest an UI enhancement to view the history about completed runs in each cron workflow.

<br>

**Example**
image
If You click a cron-workflow, 

image
You can look history of completed runs about the cron-workflows.

<br>

# Use Cases

It is easy to view completed runs of each cron workflow.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by yunwoo-yu**
Fixes #11706 

### Motivation

- You can view completed workflows on the cron-workflows detail page.
- You can explore workflows that you want users to see more easily. Better UX
- We've taken the component that was applied to the workflow view and reused it.

### Modifications

- before
image

- after
image

### CODE REVIEW

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 1]
```diff
@@ -62,6 +73,16 @@ export const CronWorkflowDetails = ({match, location, history}: RouteComponentPr
 
     useEffect(() => setEdited(true), [cronWorkflow]);
 
+    useEffect(() => {
+        services.workflows.list(namespace, selectedPhases, selectedLabels, pagination).then(res => {
```
Comment (agilgur5) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:77]: Ideally we can filter in the request to not retrieve unnecessary workflows, although that might not be possible

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 2]
```diff
@@ -25,6 +30,12 @@ export const CronWorkflowDetails = ({match, location, history}: RouteComponentPr
     const [name] = useState(match.params.name);
     const [sidePanel, setSidePanel] = useState(queryParams.get('sidePanel'));
     const [tab, setTab] = useState(queryParams.get('tab'));
+    const [workflows, setWorkflows] = useState<Workflow[]>([]);
+    const [{selectedPhases, selectedLabels, paginationLimit}, setSelectedData] = useState<WorkflowListRenderOptions>(JSON.parse(localStorage.getItem('ListOptions/options')));
```
Comment (agilgur5) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:34]: there are no phases or labels here. those are from the sidebar filter and the Workflows list which is not present here.

pagination does not seem to be present either. there seems to be some logic here, but no UI elements for pagination.

also the `localStorage` call would not apply here as it is specific to the Workflows list page

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 3]
```diff
@@ -183,6 +204,17 @@ export const CronWorkflowDetails = ({match, location, history}: RouteComponentPr
 
         return items;
     };
+    const updateCurrentlySelectedAndBatchActions = (newSelectedWorkflows: Map<string, Workflow>): void => {
```
Comment (agilgur5) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:207]: So the checkbox is unused here. No batch actions were added to this page, so it's not useable currently

**Review by agilgur5 — CHANGES_REQUESTED** [review]
Thanks for implementing this!

There are some issues with the implementation right now, as not everything is transferable from the Workflows List. See some comments below

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx (hunk 1 above)
Comment (agilgur5) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:77]: this is possible -- you can pass in a label with the Cron Workflow name: `workflows.argoproj.io/cron-workflow=<name-of-cron-workflow>`

Comment (agilgur5) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:77]: also should use async/await per #11740

Comment (yunwoo-yu) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:77]: - Thanks for the feedback, would it be ok to change to async/await and get the data via label names like on the workflows List page?

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx (hunk 2 above)
Comment (yunwoo-yu) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:34]: - Thanks I was wondering this, would it be a good idea to put a selection or pagination in the detail cron page?

**Review by yunwoo-yu — COMMENTED** [review]
Thanks for the feedback, I've got a few questions and will try to revise it in the direction of your feedback 👍🙇‍♂️

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx (hunk 3 above)
Comment (yunwoo-yu) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:207]: - The `workflows-row` component has fixed selection and checkboxes. What do you think about creating a new `cron-workflws-row` component?

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx (hunk 1 above)
Comment (agilgur5) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:77]: You can replace `selectedLabels` (the 3rd argument to [`list`](https://github.com/argoproj/argo-workflows/blob/477b3caf415d1f65f71dd366d9ebc5c04c64c099/ui/src/app/shared/services/workflows-service.ts#L42)) with the `cron-workflow` label from above

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx (hunk 2 above)
Comment (agilgur5) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:34]: I don't think filtering is needed right now. pagination would be nice to have, but for a first-pass MVP, I think we can leave out pagination as well.

You may have not noticed, but the Cron Workflows page itself actually doesn't implement pagination yet 😅 : #10846

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx (hunk 3 above)
Comment (agilgur5) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:207]: So ideally we would factor out a `workflows-row` without checkboxes and then use that in both places.
In particular, this history view you've created could also be useful for Workflow Templates as well, so making it reusable would be useful.

That being said, that requires a bit of refactoring. As a first-pass, we could just make the checkboxes no-op for now and refactor later -- that was why I commented on the function itself and not the UI elements.
I'll leave that decision up to you for if you want to try to tackle a refactor or just leave the checkboxes non-functional for now.

**Comment by yunwoo-yu** [comment]
@agilgur5

hello 😄

I'll hide the checkbox for now, thanks for the advice

The PR got stuck in the commit log, so I'll repromote it again

Thanks!

**Comment by agilgur5** [comment]
> The PR got stuck in the commit log, so I'll repromote it again

You can rebase the branch and force push. No need to open a new PR

**Comment by yunwoo-yu** [comment]
Uncontributed commits from collaborators got mixed up in the middle, and it was really hard to pick them out 😅

And I deleted the already forked repository. Maybe juijeong8324 will PR it again!

**Comment by agilgur5** [comment]
@yunwoo-yu I noticed in this PR and in #11788 that you used your fork's `master` branch. You usually want to use [feature branches](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow). That makes it easier to separate your work and could make it easier to rebase and cherry-pick as well.

**Comment by yunwoo-yu** [comment]
@agilgur5 

For the next PR, I'll create a separate feat branch and upload it. Thanks 😁

And like you said, I was able to get rid of the merged fact with reflog. I had already deleted the repository so I didn't get to try it, but I'll do it the way you suggested next time. 👍

**Pull request closed without merging**